src/overlays/actors/ovl_Obj_Rotlift/z_obj_rotlift.c

```diff
@@ -28,24 +36,108 @@ const ActorInit Obj_Rotlift_InitVars = {
+            curDekuFlower->dyna.actor.world.pos.y = this->dyna.actor.world.pos.y + 380.0f * this->dyna.actor.scale.y;
+            curDekuFlower->dyna.actor.world.pos.z =
+                this->dyna.actor.world.pos.z + posOffset * Math_CosS(this->dyna.actor.shape.rot.y);
+            curDekuFlower->dyna.actor.shape.rot.y = this->dyna.actor.shape.rot.y;
+        }
+        posOffset -= 600.0f;
+    }
+}
+
+void ObjRotlift_Init(Actor* thisx, GlobalContext* globalCtx2) {
+    GlobalContext* globalCtx = globalCtx2;
+    ObjRotlift* this = THIS;
+    s32 type = OBJROTLIFT_GET_TYPE(&this->dyna.actor);
+    s32 dekuFlowerType;
+    s32 i;
+    ModelInfo* sModelInfo;
+    ObjEtcetera** dekuFlowers;
+
+    Actor_ProcessInitChain(&this->dyna.actor, sInitChain);
+    if (type == 0) {
+        for (dekuFlowers = this->dekuFlowers, i = 0; i < ARRAY_COUNT(this->dekuFlowers); i++, dekuFlowers++) {
+            if (!OBJROTLIFT_GET_4000(thisx) || (i != 0)) {
+                dekuFlowerType = 0;
+            } else {
+                dekuFlowerType = 0x100;
+            }
+            *dekuFlowers = (ObjEtcetera*)Actor_SpawnAsChild(
+                &globalCtx->actorCtx, &this->dyna.actor, globalCtx, ACTOR_OBJ_ETCETERA, this->dyna.actor.world.pos.x,
+                this->dyna.actor.world.pos.y, this->dyna.actor.world.pos.z, this->dyna.actor.shape.rot.x,
+                this->dyna.actor.shape.rot.y, this->dyna.actor.shape.rot.z, dekuFlowerType);
+        }
+        ObjRotlift_MoveDekuFlowers(this);
+    }
+    DynaPolyActor_Init(&this->dyna, 3);
+
+    sModelInfo = &D_80B96160[type];
+    DynaPolyActor_LoadMesh(globalCtx, &this->dyna, sModelInfo->colHeader);
+    sModelInfo->animMat = Lib_SegmentedToVirtual(sModelInfo->animMat);
+}
 
-extern InitChainEntry D_80B96178[];
+void ObjRotlift_Destroy(Actor* thisx, GlobalContext* globalCtx) {
+    ObjRotlift* this = THIS;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Obj_Rotlift/func_80B95E20.s")
+    DynaPoly_DeleteBgActor(globalCtx, &globalCtx->colCtx.dyna, this->dyna.bgId);
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Obj_Rotlift/ObjRotlift_Init.s")
+void ObjRotlift_Update(Actor* thisx, GlobalContext* globalCtx) {
+    ObjRotlift* this = THIS;
+    s16 angShift;
+    s32 angVelocity;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Obj_Rotlift/ObjRotlift_Destroy.s")
+    if (OBJROTLIFT_GET_TYPE(&this->dyna.actor) == 0) {
+        ObjRotlift_MoveDekuFlowers(this);
+    }
+    if (thisx->params >= 0) {
+        angVelocity = -0xC8;
+    } else {
+        angVelocity = 0xC8;
+    }
+    angShift = angVelocity;
+    this->dyna.actor.shape.rot.y += angShift;
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Obj_Rotlift/ObjRotlift_Update.s")
+void ObjRotlift_Draw(Actor* thisx, GlobalContext* globalCtx) {
+    s32 pad;
+    ObjRotlift* this = THIS;
+    ModelInfo* sModelInfo;
+    sModelInfo = &D_80B96160[OBJROTLIFT_GET_TYPE(&this->dyna.actor)];
```

Comment:
```suggestion
    ModelInfo* modelInfo = &sModelInfo[OBJROTLIFT_GET_TYPE(&this->dyna.actor)];
```
Can combine definition and assignment into one line. The function var isn't static so should not be prefixed with an `s`, however the static `D_80B96160` makes sense as `sModelInfo`.